api/types/resource.go

```diff
@@ -264,3 +267,34 @@ var validLabelKey = regexp.MustCompile(LabelPattern)
 func IsValidLabelKey(s string) bool {
 	return validLabelKey.MatchString(s)
 }
+
+// MatchSearch goes through select field values from a resource
+// and tries to match against the list of search values.
+// Returns true if all search vals were matched (or if nil search vals).
+// Returns false if no or partial match (or nil field values).
+func MatchSearch(fieldVals []string, searchVals []string, customMatch func(val string) bool) bool {
+	for _, searchV := range searchVals {
+		foundMatch := false
+
+		// Iterate through field values to look for a match.
+		for _, fieldV := range fieldVals {
+			if utils.ContainsIgnoreCase(fieldV, searchV) {
```

Comment:
This will ultimately call `ToLower` on the strings in `fieldVals` repeatedly. Perhaps they should be case-folded once at the beginning, or maybe there's some smarter case-folded substring matcher?